File: backend/Services/Parsers/NerdleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class NerdleParser : BasicScoreResultParser
    {
        private ILogger<NerdleParser> _logger;

        public NerdleParser(ILogger<NerdleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override string Category => "Other";
        public override string GameName => "Nerdle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName.ToLower()}game \\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => $"{Url} #nerdle";
        public override string Url => "https://nerdlegame.com";
    }
}

File: lambdas/src/parsers/nerdle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class NerdleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Nerdle';
  readonly helpText = null;
  protected readonly extraContent = `${this.url} #nerdle`;
  readonly url = 'https://nerdlegame.com';
  
  // Matches "nerdlegame 827 3/6"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName.toLowerCase()}game \\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
